cli/spinner.ts

```diff
@@ -0,0 +1,163 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+const encoder = new TextEncoder();
+
+const LINE_CLEAR = encoder.encode("\r\u001b[K"); // From cli/prompt_secret.ts
+const COLOR_RESET = "\u001b[0m";
+const DEFAULT_SPEED = 75;
+
+/** The options for the spinner */
+export interface SpinnerOptions {
+  /** The spinner to be animated */
+  spinner: string[];
+  /** The message to display next to the spinner */
+  message?: string;
+  /** The speed of the spinner. Defaults to 75 */
+  speed?: number;
+  /** The color of the spinner. Defaults to the default terminal color */
+  color?: string;
+  /** Whether to cleanup the output if the process is terminated. Defaults to true */
+  cleanup?: boolean;
+}
+
+/** The default spinners */
+export const spinners = {
+  lines: ['-', '\\', '|', '/'],
+  binary: ['0000', '1000', '1100', '1110', '1111', '0111', '0011', '0001'],
+  dots: ['⠋', '⠙', '⠹', '⠸', '⠼', '⠴', '⠦', '⠧', '⠇', '⠏'],
+  dots2: ['⣾', '⣽', '⣻', '⢿', '⡿', '⣟', '⣯', '⣷'],
+  blink: ['◜', '◠', '◝', '◞', '◡', '◟'],
+}
+
```

Comment:
I think we should provide a single reasonable default spinner. The `dots` spinner LGTM because it's the one Homebrew uses (my opinion).